servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -213,47 +202,38 @@ private static String absoluteToRelativeFormRequestTarget(final String requestTa
             final int questionMarkIdx = requestTarget.indexOf('?', fromIndex);
             return questionMarkIdx < 0 ? "/" : '/' + requestTarget.substring(questionMarkIdx);
         }
-
-        @Override
-        public Completable closeAsync() {
-            // Make a best effort to clear the map. Note that we don't attempt to resolve race conditions between
-            // closing the client and in flight requests adding Keys to the map. We also don't attempt to remove
-            // from the map if a request fails, or a request is made after the client is closed.
-            return new SubscribableCompletable() {
-                @Override
-                protected void handleSubscribe(final Subscriber subscriber) {
-                    urlKeyCache.clear();
-                    deliverCompleteFromSource(subscriber);
-                }
-            };
-        }
     }
 
     private static final class UrlKey {
         final String scheme;
-        final HostAndPort hostAndPort;
+        final String host;
+        final int port;
+        private final int hashCode;
 
-        UrlKey(final String scheme, final HostAndPort hostAndPort) {
+        UrlKey(final String scheme, final String host, final int port) {
             this.scheme = scheme;
-            this.hostAndPort = hostAndPort;
+            this.host = host;
+            this.port = port;
+            // hashCode is required at least once, but may be necessary multiple times for a single selectClient run
+            this.hashCode = 31 * (caseInsensitiveHashCode(host) + port) + scheme.hashCode();
```

Comment:
I also tried removing `31 * ` from here, but it doesn't make any difference 😄

Reply:
I'd leave it for consistency.